code/modules/crafting/craft.dm

```diff
@@ -1,4 +1,7 @@
-/datum/personal_crafting
+/datum/component/personal_crafting/Initialize()
+	RegisterSignal(parent, COMSIG_CRAFT_MENU_OPEN, .proc/ui_interact)
```

Comment:
Have the component create the button that it listens for the click on instead of having the button send the signal to the clicker